nvflare/edge/executors/edge_dispatch_executor.py

```diff
@@ -104,10 +109,12 @@ def task_received(self, task_name: str, task_data: Shareable, fl_ctx: FLContext)
         self.task_data = task_data
         self.task_sequence += 1
         self.devices = {}  # Devices got this task
-        self.num_results = 0  # Number of devices reported results
+        with self.lock:
+            self.num_results = 0  # Number of devices reported results
```

Comment:
I was actually checking if we need to guard this "self.num_results" 

Will "handle_result_report" and "task_received" method always be called by the same thread?